gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/schema/X509Scheme.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.security.service.schema;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.zuul.context.RequestContext;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.zowe.apiml.auth.Authentication;
+import org.zowe.apiml.auth.AuthenticationScheme;
+import org.zowe.apiml.gateway.security.login.x509.X509CommonNameUserMapper;
+import org.zowe.apiml.security.common.token.QueryResponse;
+
+import javax.servlet.http.HttpServletRequest;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.Base64;
+import java.util.function.Supplier;
+
+@Component
+@Slf4j
+public class X509Scheme implements AbstractAuthenticationScheme {
```

Comment:
It would be nice to document the meaning for this class and the internal one in JavaDoc.